File: Clothy.Services/Clothy.OrderService/Clothy.OrderService.Domain/Entities/OrderStatus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Clothy.OrderService.Domain.Entities
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum OrderStatus
    {
        AwaitingPayment = 0,
        Processing = 1,
        Shipped = 2,
        Delivered = 3
    }
}

File: src/Clothy.Frontend/src/entities/ordersService/orderStatus.ts
export const OrderStatus = {
    AwaitingPayment: "AwaitingPayment",
    ProcessingByManagers: "Processing",
    Shipped: "Shipped",
    Delivered: "Delivered",
} as const;

export type OrderStatus = (typeof OrderStatus)[keyof typeof OrderStatus];

export const ORDER_STATUS_OPTIONS = [
    { value: "", label: "All Statuses" },
    { value: OrderStatus.AwaitingPayment, label: "AwaitingPayment" },
    { value: OrderStatus.ProcessingByManagers, label: "Processing" },
    { value: OrderStatus.Shipped, label: "Shipped" },
    { value: OrderStatus.Delivered, label: "Delivered" },
];